subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -225,6 +225,12 @@ int bt_mesh_adv_enable(void)
 	return 0;
 }
 
+int bt_mesh_adv_disable(void)
```

Comment:
I'm little concerned about the naming here. The function says "_adv_disable()", and internally calls "_thread_suspend()". These two are slightly different operations when you look at these words - disable vs suspend. May be to avoid confusions in future, it may be nicer to have API named as "bt_mesh_adv_suspend()".

Reply:
I agree, but there is no `thread_disable()` API, and adding a `_adv_disable()` function requires adding the function both to the legacy and extended advertiser. A possible solution could be to rename both to suspend, but this might pose some issues with how we want the functionality to be with the extended advertising...